python/ql/src/semmle/python/frameworks/Django.qll

```diff
@@ -1972,4 +2028,32 @@ private module Django {
       )
     }
   }
+
+  // ---------------------------------------------------------------------------
+  // django.shortcuts.redirect
+  // ---------------------------------------------------------------------------
+  /**
+   * A call to `django.shortcuts.redirect`.
+   *
+   * Note: This works differently depending on what argument is used.
+   * _One_ option is to redirect to a full URL.
+   *
+   * See https://docs.djangoproject.com/en/3.1/topics/http/shortcuts/#redirect
+   */
+  private class DjangoShortcutsRedirectCall extends HTTP::Server::HttpRedirectResponse::Range,
+    DataFlow::CfgNode {
+    override CallNode node;
+
+    DjangoShortcutsRedirectCall() { node.getFunction() = django::shortcuts::redirect().asCfgNode() }
+
+    override DataFlow::Node getRedirectLocation() {
+      result.asCfgNode() in [node.getArg(0), node.getArgByName("to")]
+    }
```

Comment:
So what is the plan for handling calls with non-url arguments? Will that currently create a bunch of FPs? Will we try to track flow into the argument and abduct the type and the target url?

Reply:
For now: Accept that there might be FPs (which is what the current query does).

If we want to handle such FPs, we might need to write specific taint-sanitizers for this specific sink, but that's future work.

Reply:
Ok, as part of the programme to recover all lost results, this is fine. Could you add a comment, though?

Reply:
> Could you add a comment, though?

I would rather not. I mean, if the content being used as the argument is user-controlled, you can always just pass in a full URL. So while this redirect can have some different behavior, I don't see why it should be any more worthy of proper FP handling than the other redirect sinks.

If you feel a comment is needed, please suggest one :wink:

Reply:
Comment suggested. I mainly feel it might be needed because the function does not exactly return what it claims to. So for debugging in the future, that might be helpful to know. Would it be hard to actually restrict `result` to url strings, I guess we could add a concept and a type tracker for those?